deltacat/compute/compactor/compaction_session.py

```diff
@@ -139,32 +176,58 @@ def _execute_compaction_round(
     primary_keys = sorted(primary_keys)
 
     # collect cluster resource stats
+    # cluster_resources = ray.cluster_resources()
+    # logger.info(f"Total cluster resources: {cluster_resources}")
+    # logger.info(f"Available cluster resources: {ray.available_resources()}")
+    # cluster_cpus = int(cluster_resources["CPU"])
+    # logger.info(f"Total cluster CPUs: {cluster_cpus}")
+
+    # collect node group resources
+
     cluster_resources = ray.cluster_resources()
     logger.info(f"Total cluster resources: {cluster_resources}")
-    logger.info(f"Available cluster resources: {ray.available_resources()}")
-    cluster_cpus = int(cluster_resources["CPU"])
-    logger.info(f"Total cluster CPUs: {cluster_cpus}")
-
+    if pg_config: # use resource in each placement group
+        node_resource_keys=None
+        cluster_resources = pg_config[1]
+        cluster_cpus = cluster_resources['CPU']
+    elif node_group_res: # use resource in each node group
```

Comment:
Is there value in hanging onto all of this code related to node groups? Are there any benefits here that we don't get from placement groups? Should it be broken out into a separate branch to help clean the code up here a bit?

Reply:
Yeah, the code can be simplified. Node group currently is not used. It could be useful when we want to design an autoscalable placement group, where we can leverage node group to set min and max for each placement group. But let me remove it for now. As an autoscalable pg may be designed in a different way.